llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -9950,6 +9981,10 @@ void llvm::findValuesAffectedByCondition(
             InsertAffected(X);
         }
       }
+
+      if (IsAssume && HasRHSC &&
+          match(A, m_Intrinsic<Intrinsic::ctpop>(m_Value(X))))
+        AddAffected(X);
```

Comment:
do you know why adding the `AddAffected(X)` is affecting the result of `isKnownNonZero` so that the range in `@pow2_32_gtnonconst_assume` get worse?

Reply:
seems like it is the `return false` here https://github.com/dtcxzyw/llvm-project/blob/ce9f5f9bfdf5253ea502b5ddf95135b0fa4380ab/llvm/lib/Analysis/ValueTracking.cpp#L616
that causes the regression if it is changed to a continue the range is unchanged by this PR

Reply:
Fixed. Thank you!